ampd/src/solana/rpc.rs

```diff
@@ -0,0 +1,71 @@
+use std::{collections::HashMap, sync::Arc};
+
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_rpc_client_api::client_error::Result as ClientResult;
+use solana_sdk::signature::Signature;
+use solana_transaction_status::{EncodedConfirmedTransactionWithStatusMeta, UiTransactionEncoding};
+use tokio::sync::RwLock;
+pub struct RpcCacheWrapper {
+    rpc_client: RpcClient,
+    tx_cache: RwLock<HashMap<Signature, Arc<EncodedConfirmedTransactionWithStatusMeta>>>,
+}
+
```

Comment:
Humm 🤔, this would probably need to work as an LRU cache. If not, we would filling up the cache indefinitely, causing a memory leak in the entire process, as afaik handlers are only created once and used during the entire process life. I will take a look into this.

Reply:
Thganks ! That were actually the plans :smile:  , fixed at 023d8daa92d1d1c7c106512aca1d9b99683973b7